lib/common/json-schema-validator.js

```diff
@@ -35,6 +40,52 @@ function jsonSchemaValidatorFactory(
     };
 
     /**
+     * validate JSON data with given JSON schema and ignore some failure if the error matches some
+     * patterns.
+     *
+     * NOTE: this function depends on the option 'allErrors` and 'verbose' is enabled
+     *
+     * @param  {Object|String} schema - JSON schema Object or schema ref id
+     * @param  {Object} data - JSON data to be validated
+     * @param  {Regex|Array<Regex>} skipPatterns - The skip patterns
+     * @return {Boolean} true if validation passes, otherwise throw error
+     */
+    JsonSchemaValidator.prototype.validatePatternsSkipped = function (schema, data, skipPatterns) {
+        assert.equal(this.options.allErrors, true,
+                  "option 'allErrors' need be enabled for validatePatternSkipped");
+        assert.equal(this.options.verbose, true,
+                 "option 'verbose' need be enabled for validatePatternSkipped");
+
+        if (this._ajv.validate(schema, data)) {
+            return true;
+        }
+
+        var errors = this._ajv.errors;
+
+        if (skipPatterns) {
+            if (!_.isArray(skipPatterns)) {
+                skipPatterns = [skipPatterns];
+            }
+            assert.arrayOfRegexp(skipPatterns, 'skip pattern should be regexp');
+
+            errors = _.filter(errors, function(error) {
+                return !_.some(skipPatterns, function(pattern) {
```

Comment:
not AND but OR relationship, if any of pattern matches then the error is ignored.